### ISSUE TITLE
Replacing default header with header-large-dark adds horizontal padding to it

### ISSUE THREAD

**Issue by zaguiini** [WordPress/twentytwentytwo#299]
**Describe the bug**

When starting with a fresh site and replacing the default header template part with `Header -- Dark`, unexpected horizontal padding is added to the header block itself.

**To Reproduce**
1. Create a new site with TT2;
2. Highlight the `Header` block;
3. Click `Replace` on the book settings popover;
4. Select `Header -- Dark`.

Verify that some horizontal padding has been added to the header itself.

**Expected behavior**

The header should span across the page, with no outer padding.

**Screenshots**

| Editor | Live site |
| ------ | -------- |
| ![image](https://user-images.githubusercontent.com/26530524/146066155-3d01f3bc-7ffe-4cac-ab95-7cb2a5deb4dc.png) | ![image](https://user-images.githubusercontent.com/26530524/146066184-1e2ec8ef-05a8-4c04-bfd6-f8186c2145c3.png) |

**Additional context**

This doesn't happen when:
- adding the template part by itself;
- adding a new header and clicking replace.

It only happens within the default `Index` template.

**Comment by kjellr** [comment]
@zaguiini I believe this should be all set as of #291. Here's a GIF of what I see when testing against the latest in this repository: 

![headers](https://user-images.githubusercontent.com/1202812/146069855-b2c616a2-6ad8-49de-a6d4-c88fa830f87f.gif)

**Comment by zaguiini** [comment]
> @zaguiini I believe this should be all set as of #291. Here's a GIF of what I see when testing against the latest in this repository:
> 
> ![headers](https://user-images.githubusercontent.com/1202812/146069855-b2c616a2-6ad8-49de-a6d4-c88fa830f87f.gif)

Awesome. Let me try again running `trunk` this time.

**Comment by zaguiini** [comment]
The issue is indeed fixed. Thanks, @kjellr!
